tests/Controller/CRUDControllerTest.php

```diff
@@ -237,7 +237,7 @@ protected function setUp(): void
 
         $this->parameterBag = new ParameterBag();
 
-        $this->container->set('sonata.admin.pool', $this->pool);
+        $this->container->set('sonata.admin.pool.do-not-use', $this->pool);
```

Comment:
```suggestion
        $this->container->set('sonata.admin.pool', $this->pool);
```

Reply:
yeah I saw, just there, there is `do-not-use`

But why it's not finding the CRUDController ?

Reply:
The `'sonata.admin.pool'` wasn't found because it wasn't set. Removing `.do-not-use` should fix it indeed.

Now I have:
```
1) Sonata\AdminBundle\Tests\DependencyInjection\Compiler\AddDependencyCallsCompilerPassTest::testProcessSortAdmins
Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: The "sonata.admin.pool" service or alias has been removed or inlined when the container was compiled. You should either make it public, or stop using the container directly and use dependency injection instead.
```

Reply:
Is this because of 
```
$pool = $container->getDefinition('sonata.admin.pool');
```
in AddDependencyCallsCompilerPass ?

Reply:
It's because we were using 
```
$pool = $container->get('sonata.admin.pool');
```
In the test, in order to make some checks
```
        $pool = $container->get('sonata.admin.pool');
        $this->assertInstanceOf(Pool::class, $pool);
        $adminServiceIds = $pool->getAdminServiceIds();
        $adminGroups = $pool->getAdminGroups();
        $adminClasses = $pool->getAdminClasses();

        $this->assertContains('sonata_post_admin', $adminServiceIds);
        $this->assertContains('sonata_article_admin', $adminServiceIds);
        $this->assertContains('sonata_news_admin', $adminServiceIds);

        $poolDefinition = $container->getDefinition('sonata.admin.pool');

        $this->assertContains('sonata_post_admin', $poolDefinition->getArgument(1));
        $this->assertArrayHasKey('sonata_group_one', $poolDefinition->getArgument(2));
        $this->assertArrayHasKey(NewsEntity::class, $poolDefinition->getArgument(3));
```

What is the solution ?

Reply:
maybe

```php
$pool = $container->get(Pool::class);
//$this->assertInstanceOf(Pool::class, $pool); remove this line
```

Reply:
here too `$poolDefinition = $container->getDefinition('sonata.admin.pool');`

Reply:
But  `Pool::class` isn't public, is it?

Reply:
`AddDependencyCallsCompilerPassTest` should extend from `AbstractCompilerPassTestCase` to make tests easier, I'll try in `3.x`, in the meanwhile, I guess 

https://github.com/sonata-project/SonataAdminBundle/blob/a077dd9b90987043730a389e2c435705a70ccd04/tests/DependencyInjection/Compiler/AddDependencyCallsCompilerPassTest.php#L246-L267

could be changed from `$pool = $container->get('sonata.admin.pool');` to `$poolDefinition = $container->findDefinition('sonata.admin.pool');` and work with `$poolDefinition->getClass()` and `$poolDefinition->getArgument(x)`.

And for the container to not remove it, adding `$container->getCompilerPassConfig()->setRemovingPasses([]);` to the `getContainer` method.

I can try later to fix them.

Reply:
Changing `sonata.admin.pool` to `Pool::class` does fix the tests because it's public https://github.com/sonata-project/SonataAdminBundle/pull/7116/files#diff-97b3cbefcb17afe6a4336f0f1be39c328d308b6443dd80eb95c9a7739c0da2ffR56

But it could be nice to add some improvement to these tests indeed if you have time @franmomu :)